flink/src/main/scala/com/metamx/tranquility/flink/BeamSink.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.flink
+
+import com.metamx.common.scala.Logging
+import com.metamx.tranquility.tranquilizer.SimpleTranquilizerAdapter
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction
+
+/**
+  * This class provides a sink that can propagate any event type to Druid.
+  * @param beamFactory your implementation of [[BeamFactory]].
+  */
+class BeamSink[T](beamFactory: BeamFactory[T])
```

Comment:
I don't know enough about flink to know if this implementation makes sense, but as long as flink doesn't need sinks to be thread-safe then it looks good to me.